test/tests/functional/pbs_hook_timeout.py

```diff
@@ -0,0 +1,186 @@
+            starttime=start_time)
+
+        # sent hook control file
+        for h in [self.hostA, self.hostB, self.hostC]:
+            hfile = os.path.join(self.server.pbs_conf['PBS_HOME'],
+                                 "server_priv", "hooks", "test.HK")
+            if h != self.hostB:
+                exist = True
+            else:
+                exist = False
+            self.server.log_match(
+                ".*successfully sent hook file %s to %s.*" %
+                (hfile, h), max_attempts=5, interval=1,
+                regexp=True, existence=exist,
+                starttime=start_time)
+
+        # Second batch of hook update is for the *.PY files + resend of
+        # *.HK file to momB
+        self.server.log_match(
+            "Timing out previous send of mom hook updates "
+            "(send replies expected=4 received=2)", n=600,
+            max_attempts=timeout_max_attempt, interval=30,
+            starttime=start_time)
+
+        # sent hook content file
+        for h in [self.hostA, self.hostB, self.hostC]:
+            hfile = os.path.join(self.server.pbs_conf['PBS_HOME'],
+                                 "server_priv", "hooks", "test.PY")
+            if h != self.hostB:
+                exist = True
+            else:
+                exist = False
+
+            self.server.log_match(
+                ".*successfully sent hook file %s to %s.*" %
+                (hfile, h), max_attempts=3, interval=1,
+                regexp=True, existence=exist,
+                starttime=start_time)
+
+        # Now check to make sure moms have received the hook files
+        for m in [self.momA, self.momB, self.momC]:
+            if m != self.momB:
+                exist = True
+            else:
+                exist = False
+            m.log_match(
+                "test.HK;copy hook-related file request received",
+                regexp=True, max_attempts=3, interval=1,
+                existence=exist, starttime=start_time)
+
+            m.log_match(
+                "test.PY;copy hook-related file request received",
+                regexp=True, max_attempts=3, interval=1,
+                existence=exist, starttime=start_time)
+
+        # Ensure that hook send updates are retried for
+        # the *.HK and *.PY file to momB
+        self.server.log_match(
+            "Timing out previous send of mom hook updates "
+            "(send replies expected=2 received=0)", n=600,
+            max_attempts=timeout_max_attempt, interval=30,
+            starttime=start_time)
+
+        # Submit a job, it should still run
+        a = {'Resource_List.select': '3:ncpus=1',
+             'Resource_List.place': 'scatter'}
+        j1 = Job(TEST_USER)
+        j1.set_attributes({'Resource_List.select': '3:ncpus=1'})
+        j1id = self.server.submit(j1)
+
+        # Wait for the job to start running.
+        a = {ATTR_state: (EQ, 'R'), ATTR_substate: (EQ, 41})
```

Comment:
Codacy and Travis identify an error on line 175. The closing brace and paren are swapped.